hobbify/users/serializers/users.py

```diff
@@ -0,0 +1,145 @@
+""" Users serializers """
+
+# Django
+from django.conf import settings
+from django.contrib.auth import password_validation, authenticate
+from django.core.validators import RegexValidator
+
+# Django REST Framework
+from rest_framework import serializers
+from rest_framework.authtoken.models import Token
+from rest_framework.validators import UniqueValidator
+
+# Models
+from hobbify.users.models import User, Profile
+
+# Tasks
+# from hobbify.taskapp.tasks import send_confirmation_email
+
+# Serializers
+from hobbify.users.serializers.profiles import ProfileModelSerializer
+
+# Utilities
+import jwt
+
+
+class UserModelSerializer(serializers.ModelSerializer):
+    """ User model serializer """
+
+    profile = ProfileModelSerializer(read_only=True)
+
+    class Meta:
+        """ Meta class """
+
+        model = User
+        fields = (
+            'username',
+            'first_name',
+            'last_name',
+            'email',
+            'phone_number',
+            'created',
+            'modified',
+            'is_hidden',
+            'profile'
+        )
+
+
+class UserSignUpSerializer(serializers.Serializer):
+    """
+    User sign up serializer.
+    Handle sign up data validation and user/profile creation.
+    """
+
+    email = serializers.EmailField(
+        validators=[UniqueValidator(queryset=User.objects.all())]
+    )
+    username = serializers.CharField(
+        min_length=4,
+        max_length=20,
+        validators=[UniqueValidator(queryset=User.objects.all())]
+    )
+
+    # Phone number
+    phone_regex = RegexValidator(
+        regex=r'\+?1?\d{9,15}$',
+        message="Phone number must be entered in the format: +999999999. Up to 15 digits allowed."
+    )
+    phone_number = serializers.CharField(validators=[phone_regex])
+
+    # Password
+    password = serializers.CharField(min_length=8, max_length=64)
+    password_confirmation = serializers.CharField(min_length=8, max_length=64)
+
+    # Name
+    first_name = serializers.CharField(min_length=2, max_length=30)
+    last_name = serializers.CharField(min_length=2, max_length=30)
+
+    def validate(self, data):
+        """ Verify passwords match """
+        passwd = data['password']
+        passwd_conf = data['password_confirmation']
+        if passwd != passwd_conf:
+            raise serializers.ValidationError("Passwords does not match.")
+        password_validation.validate_password(passwd)
```

Comment:
does this raise an exception? If so, shouldn't you be handling it and re-raising it?